File: src/main/java/org/museautomation/desktop/runner/jobs/Job.java
package org.museautomation.desktop.runner.jobs;

/**
 * @author Christopher L Merrill (see LICENSE.txt for license details)
 */
public class Job
    {
    public String getId()
        {
        return _id;
        }

    public void setId(String id)
        {
        _id = id;
        }

    public String getTaskId()
        {
        return _task_id;
        }

    public void setTaskId(String task_id)
        {
        _task_id = task_id;
        }

    public String getProjectId()
        {
        return _project_id;
        }

    public void setProjectId(String project_id)
        {
        _project_id = project_id;
        }

    private String _id;
    private String _task_id;
    private String _project_id;
    }

File: runner-core/src/main/kotlin/org/museautomation/runner/jobs/Job.kt
package org.museautomation.runner.jobs

import org.museautomation.core.values.descriptor.SimpleSubsourceDescriptor

/**
 * @author Christopher L Merrill (see LICENSE.txt for license details)
 */
data class Job (var id: String,
                var taskId: String,
                var projectId: String,
                var inputs: List<SimpleSubsourceDescriptor>,
                var outputFile: String?,
                var outputs: List<JobOutputSpec>,
                var successMessageFormat: String?)